pyccel/ast/bitwise_operators.py

```diff
@@ -40,14 +40,15 @@ class PyccelInvert(PyccelUnaryOperator):
     _precedence = 14
     _dtype     = NativeInteger()
 
-    def _set_dtype(self):
-        a = self._args[0]
+    def _calculate_dtype(self, *_args):
+        a = _args[0]
         if a.dtype not in (NativeInteger(), NativeBool()):
-            raise TypeError('unsupported operand type(s): {}'.format(self))
+            raise TypeError('unsupported operand type(s): {}'.format(_args))
 
         self._args      = (PythonInt(a) if a.dtype is NativeBool() else a,)
-
-        self._precision = a.precision
+        _precision = a.precision
+        _dtype = a.dtype
+        return _dtype, _precision
```

Comment:
You have the wrong dtype here

Reply:
Ah sorry, you are right. The dtype is set after the casting so it is correct. The class property is no longer necessary though

Reply:
I added the class property because I saw that the `_args` may get updated `if a.dtype is NativeBool()` . I don't know if I understood it wrong ...

Reply:
The class property is the property defined at line 41

Reply:
The `dtype` is always `NativeInteger` here. If the argument is an integer then it remains the same, if it is a boolean then it is cast to an integer. So the args are updated but the final result is always an integer.

What you have done is fine. But you need to remove line 41.

Reply:
> done in cab4c95

I think this is the wrong commit message, I just see a removed print

Reply:
Ah no sorry there is a problem with the dtype, you are collecting the dtype from the original argument, not the corrected argument